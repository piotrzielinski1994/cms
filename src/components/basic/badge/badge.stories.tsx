import { getThemeConfig } from '@/config/store/themes.config';
import { StoryContext } from '@/config/storybook/components';
import { DEFAULT_VALUE, getFallback } from '@/config/storybook/utils';
import type { Meta, StoryObj } from '@storybook/react';
import { useTranslations } from 'next-intl';
import { type ComponentProps } from 'react';
import { Badge as BadgeComponent } from './badge';

type Args = ComponentProps<typeof BadgeComponent>;

const meta = {
  component: BadgeComponent,
  title: 'Basic/Badge',
  argTypes: {
    label: { control: 'text' },
    bgColor: { control: 'text' },
    textColor: { control: 'text' },
  },
  args: {
    label: DEFAULT_VALUE,
    bgColor: DEFAULT_VALUE,
    textColor: DEFAULT_VALUE,
  },
} satisfies Meta<Args>;

const Render = ({ label, bgColor, textColor, ...args }: Args, context) => {
  const t = useTranslations('storybook.basic');
  const { theme } = context.globals as StoryContext['globals'];

  return (
    <div className="flex gap-1">
      <BadgeComponent
        {...args}
        label={getFallback(label, t('badge'))}
        bgColor={getFallback(bgColor, getThemeConfig(theme).foreground)}
        textColor={getFallback(textColor, getThemeConfig(theme).background)}
      />
    </div>
  );
};

const Badge: StoryObj<typeof BadgeComponent> = { render: Render };

export { Badge };
export default meta;
