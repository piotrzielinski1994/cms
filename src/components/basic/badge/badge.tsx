import { cn } from '@/utils/tailwind';
import { PropsWithChildren } from 'react';

type BadgeProps = PropsWithChildren & {
  label: string;
  bgColor: string;
  textColor: string;
};

const Badge = ({ label, bgColor, textColor }: BadgeProps) => {
  return (
    <div
      className={cn('py-1 px-2 w-fit', 'flex gap-2 flex-wrap', 'text-xs font')}
      style={{ backgroundColor: bgColor, color: textColor }}
    >
      {label}
    </div>
  );
};

export { Badge };
