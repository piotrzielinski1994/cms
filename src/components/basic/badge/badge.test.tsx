import { render } from '@testing-library/react';
import { ComponentProps } from 'react';
import { describe, expect, it } from 'vitest';
import { axe } from 'vitest-axe';
import { Badge } from './badge';

describe('Badge', () => {
  const defaultProps = {
    bgColor: '#000',
    textColor: '#fff',
    label: 'Label',
  } satisfies ComponentProps<typeof Badge>;

  it('should have no accessibility violations', async () => {
    const { container } = render(<Badge {...defaultProps} />);
    const results = await axe(container);
    expect(results).toHaveNoViolations();
  });

  it('should match the snapshot', async () => {
    const { container } = render(<Badge {...defaultProps} />);
    expect(container).toMatchSnapshot();
  });
});
