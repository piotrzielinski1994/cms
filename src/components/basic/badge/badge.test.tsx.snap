// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Badge > should match the snapshot 1`] = `
<div>
  <div
    class="py-1 px-2 w-fit flex gap-2 flex-wrap text-xs font"
    style="background-color: rgb(0, 0, 0); color: rgb(255, 255, 255);"
  >
    Label
  </div>
</div>
`;
